pkg/cluster/bootstrap.go

```diff
@@ -478,7 +478,16 @@ func (c *Cluster) compareConfig() error {
 	if !reflect.DeepEqual(clusterControl.CriticalControlArgs, c.config.CriticalControlArgs) {
 		logrus.Debugf("This is the server CriticalControlArgs: %#v", clusterControl.CriticalControlArgs)
 		logrus.Debugf("This is the local CriticalControlArgs: %#v", c.config.CriticalControlArgs)
-		return errors.New("critical configuration value mismatch")
+
```

Comment:
This is inside a block that already does a DeepEqual on the struct as a whole. Should we just do away with the outer check, and replace it with this field-by-field comparison that outputs a more meaningful message?

I will also note that this isn't the only field where the name doesn't intuitively map to a CLI option. Rather than renaming fields, you might add a tag to the struct (similar to the json tag I suggested earlier) that denotes which CLI flag it comes from, and use that when raising a mismatch error.

Reply:
Something like:
```
EncryptSecrets           bool `cli:"--secrets-encrypt"`
```

You can access these with https://pkg.go.dev/reflect#StructTag

Reply:
So I was thinking about the simple case: most of the time, the criticalconfig args are not mismatched. So for performance and clarity reasons, doing the simple if at the top makes sense. Its only when it fails that we want to spend more time and dig deeper.
That being said, I think keeping the original struct field name and instead adding a `cli` tag is more helpful for error messages.

Reply:
> for performance and clarity reasons, doing the simple if at the top makes sense.

DeepEqual already recurses into the struct and basically does what you're doing here; I think we might as well just do it once and be more clear about it.

Reply:
Converted to using `deep.Equal` library which returns a list of differences.